src/main/java/core/basesyntax/FileEditor.java

```diff
@@ -0,0 +1,279 @@
+                }
+            } else {
+                System.out.println("Указанный путь не существует");
+            }
+        }
+    }
+
+    private static void createFile(String consoleLine, String fileData) {
+        consoleLine = consoleLine.trim();
+        int commandSeparator = consoleLine.indexOf(" ");
+
+        if (commandSeparator == -1) {
+            System.out.println("Не верный формат команды");
+        } else {
+            Path fullPath = Paths.get(consoleLine.substring(0, commandSeparator) + "\\"
+                    + consoleLine.substring(commandSeparator).trim());
+            Path path = fullPath.getParent();
+            Path fileName = fullPath.getFileName();
+
+            if (Files.notExists(path)) {
+                int answer = getAnswerYesNo("Путь \"" + path + "\" не существует. Создать?");
+
+                if (answer == 1) {
+                    try {
+                        Files.createDirectory(path);
+                    } catch (IOException e) {
+                        System.out.println("Проверьте формат пути. "
+                                + "Конечный каталог/файл не был создан");
+                    }
+                }
+            }
+
+            if (Files.exists(path)) {
+                if (Files.exists(fullPath)) {
+                    int answer = getAnswerYesNo("Файл \"" + fileName + "\" существует. "
+                            + "Разрешить перезапись?");
+
+                    if (answer == 1) {
+                        try {
+                            List<String> lines = Arrays.asList(fileData);
+                            Files.write(fullPath, lines, StandardCharsets.UTF_8,
+                                    StandardOpenOption.APPEND);
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                } else {
+                    try {
+                        Files.createFile(fullPath);
+                        if (!fileData.isEmpty()) {
+                            List<String> lines = Arrays.asList(fileData);
+                            Files.write(fullPath, lines, StandardCharsets.UTF_8,
+                                    StandardOpenOption.APPEND);
+                        }
+                    } catch (IOException e) {
+                        System.out.println("Проверьте формат команды. "
+                                + "Конечный каталог/файл не был создан");
+                    }
+                }
+            }
+        }
+    }
+
+    public static int getAnswerYesNo(String question) {
+        System.out.println(question + " (y/n)");
+
+        InputStreamReader reader = new InputStreamReader(System.in);
+        char answer = 0;
+        try {
+            answer = (char) reader.read();
+            if (answer == 'y' || answer == 'Y') {
+                return 1;
+            } else if (answer == 'n' || answer == 'N') {
+                return -1;
```

Comment:
it would be more canonically to use 0 as a negative answer. -1 is more like non valid answer